### PR TITLE
[Builtins] Drop 'Proxy' from 'TypeSchemeAll'

### PR DESCRIPTION
**Pull request by effectfully**
Do not look here yet.

### CODE REVIEW

**Comment by effectfully** [comment]
I was going to make these changes a part of a bigger refactoring, but I want to conduct an experiment and that `Proxy` in `TypeSchemeAll` gets in the way of the optimizer, so I decided to drop it now (and a couple more of them while I'm here, and the now redundant `Arity` stuff). 

This only deletes code, so I'm not waiting for a review and merging as is.

File: plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs [hunk 1]
```diff
@@ -87,26 +88,20 @@ infixr 9 `TypeSchemeArrow`
...
-           -- directly we introduce an additional and "redundant" type variable. The reason why we
-           -- do that is because this way we can also bind such a variable later when constructing
-           -- the 'TypeScheme' of a polymorphic builtin manually, so that for the user this looks
-           -- exactly like a single bound type variable instead of this weird @TyVarRep@ thing.
-        -> (forall ot. ot ~ TyVarRep ('TyNameRep @kind text uniq) =>
-               Proxy ot -> TypeScheme term args res)
         -> TypeScheme term args res
+        -> TypeScheme term args res
+
+argOf :: TypeScheme term (arg ': args) res -> Proxy arg
```
Comment (michaelpj) [inline plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs:103]: argProxy? whatever

**Review by michaelpj — COMMENTED** [review]
Nice. And you didn't even need any type applications!

**Pull request merged**